### DISCUSSION TITLE
How to disable Panic exceptions using a compiled filament library.

### DISCUSSION THREAD

**Discussion by i81343** [google/filament#4955] [Q&A] [answered]
Hi, 
I'm working on handle case when Filament's is running out of memory.
For example when creating vertexbuffers until filament runs out of memory.
In this cases filament is lauching a panic exception on debug.
I have seen that in source code there is the next explanation of how to do it from c++:

```
 * ### Testing Code that Uses Panics ###
 *
 * Since panics use exceptions for their underlying implementation, you can test code that uses
 * panics with EXPECT_THROW by doing the following things:
 * \li Set panic mode to THROW (default is TERMINATE)
 * \li Pass Panic to EXPECT_THROW as the exception type
 *
 * Example code for your test file:
 *
 * @code
 * #include <MyClass.hpp>  // since your code uses panics, this should include utils/Panic.hpp
 *
 * using utils::Panic;
 *
 * TEST(MyClassTest, value_that_causes_panic) {
 *     EXPECT_THROW(MyClass::function(value_that_causes_panic), Panic);
 * }
 *
 * // ... other tests ...
 *
 * int main(int argc, char** argv) {
 *     ::testing::InitGoogleTest(&argc, argv);
 *     Panic::setMode(Panic::Mode::THROW);
 *     return RUN_ALL_TESTS();
 * }
 * @endcode
 *
 */
```
 However if I use a compiled library, in the 'public' includes, is not included "utils/Panic.hpp" therefore i can't disable panic exceptions using a library.
 
How can I use it so that disable Panic expections?

Thanks in advance.

**Comment by bejado** [comment, marked as answer]
On mobile platforms panic exceptions should already be disabled, since we build with `-fno-exceptions`. For desktop builds you'll need to modify the root [`CMakeLists`](https://github.com/google/filament/blob/034ebe5fd15e1682bf1ee3b0ca883b75bca3e5d7/CMakeLists.txt#L336-L340) to disable exceptions.
